plugin/advancedsubscription/ajax/advsub.ajax.php

```diff
@@ -0,0 +1,248 @@
+            try {
+                $res = AdvancedSubscriptionPlugin::create()->isAbleToRequest($data['u'], $data);
+                if ($res) {
+                    $result['error'] = false;
+                    $result['errorMessage'] = 'No error';
+                    $result['pass'] = true;
+                } else {
+                    $result['errorMessage'] = 'User can not be subscribed';
+                    $result['pass'] = false;
+                }
+            } catch (\Exception $e) {
+                $result['errorMessage'] = $e->getMessage();
+            }
+            break;
+        case 'subscribe': // Subscription
+            // Start subscription to queue
+            $res = AdvancedSubscriptionPlugin::create()->startSubscription($data['u'], $data['s'], $data);
+            // Check if queue subscription was successful
+            if ($res === true) {
+                // Prepare data
+                // Get session data
+                $sessionArray = api_get_session_info($data['s']);
+                $extraSession = new ExtraFieldValue('session');
+                $var = $extraSession->get_values_by_handler_and_field_variable($data['s'], 'description');
+                $sessionArray['description'] = $var['field_valiue'];
+                $var = $extraSession->get_values_by_handler_and_field_variable($data['s'], 'target');
+                $sessionArray['target'] = $var['field_valiue'];
+                $var = $extraSession->get_values_by_handler_and_field_variable($data['s'], 'mode');
+                $sessionArray['mode'] = $var['field_valiue'];
+                $var = $extraSession->get_values_by_handler_and_field_variable($data['s'], 'publication_end_date');
+                $sessionArray['publication_end_date'] = $var['field_value'];
+                $var = $extraSession->get_values_by_handler_and_field_variable($data['s'], 'recommended_number_of_participants');
+                $sessionArray['recommended_number_of_participants'] = $var['field_valiue'];
+                // Get student data
+                $studentArray = api_get_user_info($data['u']);
+                $studentArray['picture'] = UserManager::get_user_picture_path_by_id($studentArray['user_id'], 'web', false, true);
+                $studentArray['picture'] = UserManager::get_picture_user($studentArray['user_id'], $studentArray['picture']['file'], 22, USER_IMAGE_SIZE_MEDIUM);
+                // Get superior data if exist
+                $superiorId = UserManager::getStudentBoss($data['u']);
+                if (!empty($superiorId)) {
+                    $superiorArray = api_get_user_info($superiorId);
+                } else {
+                    $superiorArray = null;
+                }
+                // Get admin data
+                $adminsArray = UserManager::get_all_administrators();
+                foreach ($adminsArray as &$admin) {
+                    $admin['complete_name'] = $admin['lastname'] . ', ' . $admin['firstname'];
```

Comment:
This should use the internationalization methods get_order_name() or something like that. Pleas check.